addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CommandInjectionScanRule.java

```diff
@@ -208,12 +209,18 @@ public class CommandInjectionScanRule extends AbstractAppParamPlugin {
         NIX_OS_PAYLOADS.put("&&" + insertedCMD + NULL_BYTE_CHARACTER, NIX_CTRL_PATTERN);
     }
 
-    // Coefficient used for a time-based query delay checking (must be >= 7)
-    private static final int TIME_STDEV_COEFF = 7;
     /** The default number of seconds used in time-based attacks (i.e. sleep commands). */
     private static final int DEFAULT_TIME_SLEEP_SEC = 5;
-    // Standard deviation limit in milliseconds (long requests deviate from a correct model)
-    public static final double WARN_TIME_STDEV = 0.5 * 1000;
+
+    // time-based attack detection will not send more than the following number of requests
+    private static final int BLIND_REQUEST_LIMIT = 5;
+    // time-based attack detection will try to take less than the following number of seconds
+    // note: detection of this length will generally only happen in the positive (detecting) case.
+    private static final double BLIND_SECONDS_LIMIT = 20.0;
```

Comment:
What happens if the user sets the sleep time to a higher value?

Reply:
Currently, this value is unaffected by the user-configured limit. The reason why I didn't immediately connect this value to that user-configured value is because I believe the user-configured limit defaults to around 5 seconds, while a limit of at least 15 seconds is minimally required before the blind algorithm will work reliably. Thus, either this should be based off of a different value, or the default value should be changed. I wanted to leave that decision up to you if possible.

To answer the question, if the `BLIND_SECONDS_LIMIT` is set higher, this generally won't affect the negative case, but most positive cases will take longer, up to a certain point. The additional BLIND_REQUEST_LIMIT puts a hard cap on how many requests can be sent, and as long as that stays at 5, the number of seconds waiting will not increase without bound.

Reply:
Sounds good and thank you for the thorough explanation.